src/components/LayerUI.tsx

```diff
@@ -173,44 +180,50 @@ const LayerUI = ({
                       />
                     </Stack.Row>
                   </Island>
-                  <LockIcon
-                    checked={appState.elementLocked}
-                    onChange={onLockToggle}
-                    title={t("toolBar.lock")}
-                  />
+                  {!showZenMode && (
+                    <LockIcon
+                      checked={appState.elementLocked}
+                      onChange={onLockToggle}
+                      title={t("toolBar.lock")}
+                    />
+                  )}
                 </Stack.Row>
               </Stack.Col>
             )}
           </Section>
           <div />
         </div>
-        <div className="App-menu App-menu_bottom">
-          <Stack.Col gap={2}>
-            <Section heading="canvasActions">
-              <Island padding={1}>
-                <ZoomActions
-                  renderAction={actionManager.renderAction}
-                  zoom={appState.zoom}
-                />
-              </Island>
-              {renderEncryptedIcon()}
-            </Section>
-          </Stack.Col>
-        </div>
+        {!showZenMode && (
```

Comment:
Instead of not rendering on the DOM, could you tweak CSS so that it is just outside of the viewport. This way you can add css transitions in order to make them move in and out of the screen. It's going to be a less jarring experience than for it to disappear all at once.

Reply:
I had removed the elements from DOM since it was not needed when using Zen mode. I think the main reason behind jarring experience was because the shapes toolbar was moving when the lock icon was disappearing, I have fixed that now so I don't feel the jarring experience now. And I tried hiding the top left canvas actions with CSS transitions to compare.
I am not sure which one is better as now the toolbar doesn't move after the lock icon is removed so need opinions @vjeux @dwelle @kbariotis can you check once which one is looking better and I can make the changes accordingly.
However even after setting visibility hidden the actions are still triggered when you hover over that area so I will check why that is happening.

Reply:
Also, I have applied transform to hide the zoom menu from the bottom so we have 3 things to compare now
I feel zoom one is a little distracting as it slides off.